manim/config.py

```diff
@@ -1,283 +1,406 @@
+                'LEAVE_PROGRESS_BARS', 'WRITE_ALL', 'WRITE_TO_MOVIE',
+                'SAVE_LAST_FRAME', 'DRY_RUN', 'SAVE_PNGS', 'SAVE_AS_GIF']:
+        config[opt] = config_parser['CLI'].getboolean(opt)
 
-    config["skip_animations"] = any([
-        file_writer_config["save_last_frame"],
-        config["start_at_animation_number"],
+    # numbers
+    for opt in ['FROM_ANIMATION_NUMBER', 'UPTO_ANIMATION_NUMBER',
+                'BACKGROUND_OPACITY']:
+        config[opt] = config_parser['CLI'].getint(opt)
+
+    # UPTO_ANIMATION_NUMBER is special because -1 actually means np.inf
+    if config['UPTO_ANIMATION_NUMBER'] == -1:
+        import numpy as np
+        config['UPTO_ANIMATION_NUMBER'] = np.inf
+
+    # strings
+    for opt in ['PNG_MODE', 'MOVIE_FILE_EXTENSION', 'MEDIA_DIR',
+                'OUTPUT_FILE', 'VIDEO_DIR', 'TEX_DIR', 'TEXT_DIR',
+                'BACKGROUND_COLOR']:
+        config[opt] = config_parser['CLI'][opt]
+
+    # streaming section -- all values are strings
+    config['STREAMING'] = {}
+    for opt in ['LIVE_STREAM_NAME', 'TWITCH_STREAM_KEY',
+                'STREAMING_PROTOCOL', 'STREAMING_PROTOCOL', 'STREAMING_IP',
+                'STREAMING_PORT', 'STREAMING_PORT', 'STREAMING_CLIENT',
+                'STREAMING_CONSOLE_BANNER']:
+        config['STREAMING'][opt] = config_parser['streaming'][opt]
+
+    # for internal use (no CLI flag)
+    config['SKIP_ANIMATIONS'] = any([
+        config['SAVE_LAST_FRAME'],
+        config['FROM_ANIMATION_NUMBER'],
     ])
-    return config
 
+    # camera config -- all happen to be integers
+    config['CAMERA_CONFIG'] = {}
+    for opt in ['FRAME_RATE', 'PIXEL_HEIGHT', 'PIXEL_WIDTH']:
+        config['CAMERA_CONFIG'][opt] = config_parser['CLI'].getint(opt)
 
-def get_camera_configuration(args):
-    camera_config = {}
-    if args.low_quality:
-        camera_config.update(constants.LOW_QUALITY_CAMERA_CONFIG)
-    elif args.medium_quality:
-        camera_config.update(constants.MEDIUM_QUALITY_CAMERA_CONFIG)
-    elif args.high_quality:
-        camera_config.update(constants.HIGH_QUALITY_CAMERA_CONFIG)
-    elif args.four_k:
-        camera_config.update(constants.FOURK_CAMERA_CONFIG)
-    else:
-        camera_config.update(constants.PRODUCTION_QUALITY_CAMERA_CONFIG)
+    # file writer config -- just pull them from the overall config for now
+    config['FILE_WRITER_CONFIG'] = {key: config[key] for key in [
+        "WRITE_TO_MOVIE", "SAVE_LAST_FRAME", "SAVE_PNGS",
+        "SAVE_AS_GIF", "PNG_MODE", "MOVIE_FILE_EXTENSION",
+        "OUTPUT_FILE", "INPUT_FILE"]}
+
+    return config, config_parser
+
+
+def _parse_cli():
+    parser = argparse.ArgumentParser(
+        description='Animation engine for explanatory math videos',
+        epilog='Made with <3 by the manim community devs'
+    )
+    parser.add_argument(
+        "file",
+        help="path to file holding the python code for the scene",
+    )
+    parser.add_argument(
+        "--scene_names",
```

Comment:
It's a bit unnatural to pass a required argument as a flag, and this also breaks backwards compatibility. Is is possible to still pass the scenes on the command line?

Reply:
yeah I think scene names should be allowed as pos args as well

Reply:
Fixed in the latest commit.